controller/basecontroller.go

```diff
@@ -24,7 +24,8 @@ func (bc *BaseController) Init(w http.ResponseWriter, r *http.Request, session d
 // will be returned. If not, then an empty string is returned ("").
 func checkForId(url *url.URL) string {
 	base := path.Base(url.RequestURI())
-	if url.EscapedPath() != "/skills" && url.EscapedPath() != "/skills/" {
```

Comment:
Let's extract this into another function.  The logic is already starting to get difficult to read.